[CIAPP] Improve Agent installation instructions for tests (#11107)
DataDog/documentation: content/en/continuous_integration/setup_tests/agent.md

          - 8126:8126
        env:
          DD_API_KEY: ${{ secrets.DD_API_KEY }}
          DD_INSIDE_CI: "true"
          DD_HOSTNAME: "none"
    steps:
      - run: make test
{{< /code-block >}}

Add your [Datadog API key][2] to your [project secrets][3] with the key `DD_API_KEY`.

[1]: https://docs.github.com/en/actions/guides/about-service-containers
[2]: https://app.datadoghq.com/account/settings#api
[3]: https://docs.github.com/en/actions/reference/encrypted-secrets
{{% /tab %}}
{{% tab "CircleCI" %}}

> To run the Agent in CircleCI, launch the agent container before running tests by using the [datadog/agent CircleCI orb][1], and stop it after to ensure results are sent to Datadog.

Reviewer: ```suggestion
To run the Agent in CircleCI, launch the Agent container before running tests by using the [datadog/agent CircleCI orb][1], and stop it after to ensure results are sent to Datadog.
```

{{% /tab %}}
{{< /tabs >}}

### Using Docker Compose

Regardless of which CI provider you use, if tests are run using [Docker Compose][5], the Datadog Agent can be run as one service:

{{< code-block lang="yaml" filename="docker-compose.yml" >}}
version: '3'
services:
  datadog-agent:
    image: "datadog/agent:latest"
    environment:
      - DD_API_KEY
      - DD_INSIDE_CI=true
      - DD_HOSTNAME=none
    ports:
      - 8126/tcp

  tests:
    build: .
    environment:
      - DD_AGENT_HOST=datadog-agent
{{< /code-block >}}

Alternatively, share the same network namespace between the Agent container and the tests container:

{{< code-block lang="yaml" filename="docker-compose.yml" >}}
version: '3'
services:
  datadog-agent:
    image: "datadog/agent:latest"
    environment:
      - DD_API_KEY
      - DD_INSIDE_CI=true
      - DD_HOSTNAME=none

  tests:
    build: .
    network_mode: "service:datadog-agent"
{{< /code-block >}}

In this case, `DD_AGENT_HOST` is not required because it is `localhost` by default.

Then, run your tests by providing your [Datadog API key][4] in the `DD_API_KEY` environment variable:

{{< code-block lang="bash" >}}

> DD_API_KEY=<MY_API_KEY> docker-compose up \

Reviewer: ```suggestion
DD_API_KEY=<YOUR_DD_API_KEY> docker-compose up \
```

  --build --abort-on-container-exit \
  tests
{{< /code-block >}}

> **Note:** In this case, also pass along all the required CI provider environment variables so build information can be attached to each test result, as described in [Tests in containers][6].

Reviewer: ```suggestion
**Note:** In this case, add all the required CI provider environment variables so build information can be attached to each test result, as described in [Tests in containers][6].
```
